modules/openapi-generator/src/main/resources/csharp/api.mustache

```diff
@@ -403,7 +403,7 @@ namespace {{packageName}}.{{apiPackage}}
             {{/required}}
             {{/formParams}}
             {{#bodyParam}}
-            localVarRequestOptions.Data = {{paramName}};
+            localVarRequestOptions.Data = {{packageName}}.Client.ClientUtils.Serialize({{paramName}});
```

Comment:
it's serialized later in ApiClient.cs for "JSON" related payload: https://github.com/OpenAPITools/openapi-generator/blob/4e24cf53f5492caee31566d13766b78137515901/modules/openapi-generator/src/main/resources/csharp/ApiClient.mustache#L157

does your content type ends with `json`?